test/random.jl

```diff
@@ -370,11 +372,11 @@ for rng in ([], [MersenneTwister(0)], [RandomDevice()])
     for f! in [rand!, randn!, randexp!]
         for T in (f! === rand! ? types : f! === randn! ? cftypes : ftypes)
             X = T == Bool ? T[0,1] : T[0,1,2]
-            for A in (Array{T}(5), Array{T}(2, 3))
+            for A in [Array{T}(5), Array{T}(2, 3), GenericArray{T}(5), GenericArray{T}(2, 3)]
```

Comment:
Ok, will do (but I would feel uncomfortable if this could happen that one array gets promoted to another type in this case...)